src/main/java/io/confluent/idesidecar/restapi/messageviewer/SimpleConsumer.java

```diff
@@ -35,47 +34,46 @@ public class SimpleConsumer {
   private static final int MAX_RESPONSE_BYTES = 20 * 1024 * 1024; // 20 MB
   private static final int DEFAULT_MESSAGE_MAX_BYTES = 4 * 1024 * 1024; // 4MB
   private final SchemaRegistryClient schemaRegistryClient;
+  private final KafkaConsumer<byte[], byte[]> consumer;
+  private final RecordDeserializer recordDeserializer;
 
-  final Properties baseConsumerConfig;
-
-  public SimpleConsumer(Properties baseConfig, SchemaRegistryClient src) {
-    schemaRegistryClient = src;
-    var props = new Properties();
-    // Custom properties
-    props.putAll(baseConfig);
-
-    // Default properties
-    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
-        "org.apache.kafka.common.serialization.ByteArrayDeserializer");
-    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
-        "org.apache.kafka.common.serialization.ByteArrayDeserializer");
-    baseConsumerConfig = props;
+  public SimpleConsumer(
+      KafkaConsumer<byte[], byte[]> consumer,
+      SchemaRegistryClient sr,
+      RecordDeserializer recordDeserializer
+  ) {
+    this.consumer = consumer;
+    this.schemaRegistryClient = sr;
+    this.recordDeserializer = recordDeserializer;
   }
 
   @SuppressWarnings("CyclomaticComplexity")
-  public List<PartitionConsumeData> consumeFromMultiplePartitions(
+  public List<PartitionConsumeData> consume(
       String topicName,
-      Map<Integer, Long> offsets,
-      Boolean fromBeginning,
-      Long timestamp,
-      Integer maxPollRecords,
-      Integer fetchMaxBytes,
-      Integer messageMaxBytes
-  ) {
-    messageMaxBytes = messageMaxBytes == null ? DEFAULT_MESSAGE_MAX_BYTES : messageMaxBytes;
+      ConsumeRequest request) {
+    final var messageMaxBytes = Optional
+        .ofNullable(request.messageMaxBytes())
+        .orElse(DEFAULT_MESSAGE_MAX_BYTES);
+
     // maxPollRecords is the number of messages we get for this entire query.
-    int recordsLimit = Math.min(
-        Optional.ofNullable(maxPollRecords).orElse(MAX_POLL_RECORDS_LIMIT),
+    var recordsLimit = Math.min(
```

Comment:
Just curious: why change from `int` to `var`?

Reply:
I've made changes across to use `var` instead of explicit datatypes wherever possible. Why? A combination of personal preference and more importantly, for consistency with the usage of `var` elsewhere in the codebase.